src/main/java/com/google/devtools/build/lib/rules/java/JavaInfo.java

```diff
@@ -424,6 +424,12 @@ private JavaInfoProvider() {
       super(STARLARK_NAME, JavaInfo.class);
     }
 
+    private void checkSequenceOfJavaInfo(Sequence<?> seq, String field) throws EvalException {
+      if (seq.stream().anyMatch(v -> !(v instanceof JavaInfo))) {
```

Comment:
opt: consider not using Streams. It's some amount of GC churn (or maybe I'm just superstitious, thus the "opt" tag)

Reply:
Not entirely superstitious: https://www.beyondjava.net/performance-java-8-lambdas

Reply:
Why can't we have nice things. :(

Reply:
I'm sorry :(